src/SonarQube.Client/Api/SonarQubeService.cs

```diff
@@ -0,0 +1,252 @@
+        }
+
+        public async Task<SonarQubeQualityProfile> GetQualityProfileAsync(string projectKey, string organizationKey, SonarQubeLanguage language, CancellationToken token)
+        {
+            var qualityProfiles = await InvokeRequestAsync<IGetQualityProfilesRequest, SonarQubeQualityProfile[]>(
+                request =>
+                {
+                    request.ProjectKey = projectKey;
+                    request.OrganizationKey = organizationKey;
+                },
+                token);
+
+            // Consider adding the language filter to the request configuration above and removing this line
+            var profilesWithGivenLanguage = qualityProfiles.Where(x => x.Language == language.Key).ToList();
+
+            var qualityProfile = profilesWithGivenLanguage.Count > 1
+                ? profilesWithGivenLanguage.Single(x => x.IsDefault)
+                : profilesWithGivenLanguage.Single();
+
+            var changeLog = await InvokeRequestAsync<IGetQualityProfileChangeLogRequest, DateTime[]>(
+                request =>
+                {
+                    request.Page = 1;
+                    request.PageSize = 1;
+                    request.QualityProfileKey = qualityProfile.Key;
+                    request.QualityProfileName = qualityProfile.Name;
+                    request.LanguageName = language.Key;
+                },
+                token);
+
+            var updatedDate = changeLog.Any() ? changeLog.Single() : qualityProfile.TimeStamp;
+
+            return new SonarQubeQualityProfile(qualityProfile.Key, qualityProfile.Name, qualityProfile.Language,
+                qualityProfile.IsDefault, updatedDate);
+        }
+
+        public async Task<RoslynExportProfileResponse> GetRoslynExportProfileAsync(string qualityProfileName,
+            string organizationKey, SonarQubeLanguage language, CancellationToken token) =>
+            await InvokeRequestAsync<IGetRoslynExportProfileRequest, RoslynExportProfileResponse>(
+                request =>
+                {
+                    request.QualityProfileName = qualityProfileName;
+                    request.LanguageKey = language.Key;
+                    request.OrganizationKey = organizationKey;
+                },
+                token);
+
+        public async Task<IList<SonarQubeIssue>> GetSuppressedIssuesAsync(string key, CancellationToken token)
+        {
+            var result = await InvokeRequestAsync<IGetIssuesRequest, SonarQubeIssue[]>(
+                request =>
+                {
+                    request.ProjectKey = key;
+                },
+                token);
+
+            return result
+                .Where(x => x.ResolutionState != SonarQubeIssueResolutionState.Unresolved)
+                .ToList();
+        }
+
+        public async Task<IList<SonarQubeNotification>> GetNotificationEventsAsync(string projectKey, DateTimeOffset eventsSince,
```

Comment:
Can projectKey be null/empty?

Reply:
If it is null or empty we will pass empty query string argument, which is ok, I think. I don't know what SQ will do in this case, though :)

Reply:
They are both required by the SQ API so presumably we'll get an error code back which will be handled as a failure by the client.
Is that the general approach that we want to take for the api calls - to leave the validation to the server? (seems ok to me)

Reply:
IMO the client should follow the API specification and if an argument is missing in a particular version of SQ it should not be sent.